### PR TITLE
Adding update policy for autoscaling group

### PR DESCRIPTION
**Pull request by amarrella**
I think that adding an update policy to the autoscaling group will make upgrading the instances much easier. This will be useful when we change instance type, amis...etc without going through manually replacing the instances.

See

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html

https://aws.amazon.com/premiumsupport/knowledge-center/auto-scaling-group-rolling-updates/

for details.

### CODE REVIEW

File: 1.0/NovaAutoScalingExternalHTTPSStack.json [hunk 1]
```diff
@@ -33,7 +33,13 @@
     "StackType": { "Type": "String" },
     "TeamName": { "Type": "String" },
     "TerminationPolicy": { "Type": "String", "Default": "Default" },
-    "VpcSubnetIds": { "Type": "CommaDelimitedList" }
+    "VpcSubnetIds": { "Type": "CommaDelimitedList" },
+    "AutoScalingRollingUpdateMaxBatchSize" : { "Type": "Integer", "Default": 1 },
+    "AutoScalingRollingUpdateMinInstancesInService" : { "Type": "Integer", "Default": 1 },
+    "AutoScalingRollingUpdateMinSuccessfulInstancesPercent" : { "Type": "Integer", "Default": 100 },
+    "AutoScalingRollingUpdatePauseTime" : { "Type": "String", "Default": "PT15M" },
+    "AutoScalingRollingUpdateSuspendProcesses" : [ "HealthCheck", "ReplaceUnhealthy", "AZRebalance", "AlarmNotification", "ScheduledActions" ],
```
Comment (gheine) [inline 1.0/NovaAutoScalingExternalHTTPSStack.json]: more of a question than a comment: is this how array type parameters with defaults are specified?

Comment (amarrella) [reply 1.0/NovaAutoScalingExternalHTTPSStack.json]: oh no, my mistake, will fix that

File: 1.0/NovaAutoScalingExternalHTTPSStack.json [hunk 2]
```diff
@@ -33,7 +33,13 @@
...
     "TeamName": { "Type": "String" },
     "TerminationPolicy": { "Type": "String", "Default": "Default" },
-    "VpcSubnetIds": { "Type": "CommaDelimitedList" }
+    "VpcSubnetIds": { "Type": "CommaDelimitedList" },
+    "AutoScalingRollingUpdateMaxBatchSize" : { "Type": "Number", "Default": 1 },
+    "AutoScalingRollingUpdateMinInstancesInService" : { "Type": "Number", "Default": 1 },
+    "AutoScalingRollingUpdateMinSuccessfulInstancesPercent" : { "Type": "Number", "Default": 100 },
+    "AutoScalingRollingUpdatePauseTime" : { "Type": "String", "Default": "PT15M" },
+    "AutoScalingRollingUpdateSuspendProcesses" : { "Type": "CommaDelimitedList", "Default":"HealthCheck,ReplaceUnhealthy,AZRebalance,AlarmNotification,ScheduledActions"},
+    "AutoScalingRollingUpdateWaitOnResourceSignals" : { "Type": "String", "Default": "true", "AllowedValues": [ "true","false"] }
```
Comment (oolekk) [inline 1.0/NovaAutoScalingExternalHTTPSStack.json]: is it possible to nest it same way as below, AutoScalingRolingUpdate: { some stuff here }, I think it's more readable and less repetition

File: 1.0/NovaAutoScalingExternalHTTP_HTTPSStack.json [hunk 3]
```diff
@@ -33,7 +33,13 @@
...
     "TeamName": { "Type": "String" },
     "TerminationPolicy": { "Type": "String", "Default": "Default" },
-    "VpcSubnetIds": { "Type": "CommaDelimitedList" }
+    "VpcSubnetIds": { "Type": "CommaDelimitedList" },
+    "AutoScalingRollingUpdateMaxBatchSize" : { "Type": "Number", "Default": 1 },
+    "AutoScalingRollingUpdateMinInstancesInService" : { "Type": "Number", "Default": 1 },
+    "AutoScalingRollingUpdateMinSuccessfulInstancesPercent" : { "Type": "Number", "Default": 100 },
+    "AutoScalingRollingUpdatePauseTime" : { "Type": "String", "Default": "PT15M" },
+    "AutoScalingRollingUpdateSuspendProcesses" : { "Type": "CommaDelimitedList", "Default":"HealthCheck,ReplaceUnhealthy,AZRebalance,AlarmNotification,ScheduledActions"},
+    "AutoScalingRollingUpdateWaitOnResourceSignals" : { "Type": "String", "Default": "true", "AllowedValues": [ "true","false"] }
```
Comment (oolekk) [inline 1.0/NovaAutoScalingExternalHTTP_HTTPSStack.json]: is it possible to nest it same way as below, AutoScalingRolingUpdate: { some stuff here }, I think it's more readable and less repetition

Comment (oolekk) [reply 1.0/NovaAutoScalingExternalHTTP_HTTPSStack.json]: same thing in other places ... not sure if it's valid, but if so, would be nice to use one way consistently

Comment (amarrella) [reply 1.0/NovaAutoScalingExternalHTTP_HTTPSStack.json]: not sure nova supports nested parameters. are they supported in cfn at all?

**Pull request merged**